packages/core/src/StateMachine.ts

```diff
@@ -428,6 +448,74 @@ export class StateMachine<
     return resolvedState as State<TContext, TEvent, TResolvedTypesMeta>;
   }
 
+  /**
+   * Creates a new machine from this machine where the initial state is at the specified `state`.
+   *
+   * @param state The state to resolve the initial state from
+   */
+  public at(
+    state?:
+      | State<TContext, TEvent, TResolvedTypesMeta>
+      | PersistedMachineState<State<TContext, TEvent, TResolvedTypesMeta>>
+  ): this {
+    if (!state) {
+      return this;
+    }
+
+    const machine = new StateMachine(this.config) as typeof this;
+    let restoredState: State<TContext, TEvent, TResolvedTypesMeta>;
+
+    if ('persisted' in state) {
+      const restoredChildren: Record<string, AnyActorRef> = {};
+
+      Object.keys(state.children).forEach((key) => {
+        const persistedState = state.children[key];
+        const impl = this.options.actors[key];
+
+        if (!impl) return;
+
+        if (typeof impl === 'function') {
+          const behavior = impl(state.context, state.event, {
+            id: key,
+            src: {} as any,
+            _event: state._event,
+            meta: undefined
+          });
+
+          const actorRef = interpret(
+            behavior.at?.(persistedState) ?? behavior,
+            { id: key }
+          );
+
+          state.actions.unshift(
+            invoke({
+              id: key,
+              // @ts-ignore TODO: fix types
+              src: actorRef, // TODO
+              ref: actorRef,
+              meta: undefined
+            })
+          );
```

Comment:
is this needed here for anything?

Reply:
Yes, this ensures that the `actorRef` actually gets invoked. Test `packages/core/test/behaviors.test.ts:386:71` fails without this.